### DISCUSSION TITLE
CEA Markov with different entry cohorts for SoC vs Intervention

### DISCUSSION THREAD

**Discussion by bzamoratalaya** [HTAhackathon2024/.github#14] [Ideas]
### What is the problem we're trying to solve

In diagnostics, the effect often changes the entry cohort for treatment, e.g., different cancer stages for early diagnosis, or targeted treatment. I have a model, published as an ISPOR poster, where I modified `{hesim}` R package to capture the early diagnosis downstaging effect.  I would like to explore different diagnostic effects of changing the comparison cohorts in the CEA model

### How will we tackle the problem?

- Define what's the effect which changes the entry cohort

- Using CEA packages and modifying the cohorts in each branch. 

- Show how is done with `{hesim}`

- Adapt to available evidence

### Which tools will we use?

`{hesim}` and others

### Previous work

[ISPOREurope23_Zamora_EE510_POSTERv1.pdf](https://github.com/user-attachments/files/16335168/ISPOREurope23_Zamora_EE510_POSTERv1.pdf)

**Comment by n8thangreen** [comment]
I've think I've done something similar before. I passed PSA samples to `{heemod}` like this
```r
res_mm <-
  heemod_init_pop_PSA(
    heemod_model,
    init_states = dt$cost$term_pop_sa)
```
Function [here](https://github.com/n8thangreen/LTBIdiagTST/blob/master/R/init_pop_heemod_PSA.R)

What format do we want the input?
Is this a case of writing a wrapper for `{heemod}` like above or something else?...

**Reply by bzamoratalaya** [reply]
@n8thangreen I'll explore how you have done in heemod. 

In the adaptation I used to the package [rcea](https://hesim-dev.github.io/rcea) and "HESIM" I did as below. This required separated analysis of the SoC cohort and the intervention/diagnostic cohorts. Finally, applying CE to the merged datasets of 2 simulated cohorts. It would be nice to avoid the step of merging datasets and integrate the package as written originally, but adding functionality of changing initial state cohorts.

##Generating the Markov trace for 5 years, we have to generate 2 different functions

##for SOC with X_init_soc


sim_stateprobsSOC <- function(p0, n_cycles){
    p <- tpmatrix(
    C,       p0$r_la,  p0$r_m,  p0$r_d,
    p0$la_r, C,             p0$la_m, p0$la_d,
    p0$m_r, p0$m_la,      C,             p0$m_d,
    0,       0,             0,             1
  )
  x_soc <- sim_markov_chain(x0 = c(0.15, 0.55, 0.3, 0),
                        p = matrix(as.matrix(p), ncol = 4, byrow = TRUE),
                        n_cycles = n_cycles)
  
  return(x_soc)
}

## For Intervention diagnostic

sim_stateprobsAD <- function(p0, n_cycles){
  p <- tpmatrix(
    C,       p0$r_la,  p0$r_m,  p0$r_d,
    p0$la_r, C,             p0$la_m, p0$la_d,
    p0$m_r, p0$m_la,      C,             p0$m_d,
    0,       0,             0,             1
  )
  
  x_AD <- sim_markov_chain(x0 = c(0.3, 0.5, 0.2, 0),
                               p = matrix(as.matrix(p), ncol = 4, byrow = TRUE),
                               n_cycles = n_cycles)
  return(x_AD)
}

- An then, I modified the package functions to create the simulation functions separate taking SOC and AD :
sim_modelSOC <- function(params_rng, n_cycles = 5, 
                      dr_qalys = .035, dr_costs = .035){........

sim_modelAD<- function(params_rng, n_cycles = 5, 
                         dr_qalys = .035, dr_costs = .035){......


- And finally I created the simulations separately for each branch and merged the data to perform the CEA

## Save results

#saveRDS(ce_simSOC,  file = "SOCexp-ce_sim.rds")
#saveRDS(ce_simAD,  file = "ADexp-ce_sim.rds")

ce_sim_part1 <- merge(ce_simSOC, ce_simAD, by.x="sample", by.y="sample", all = TRUE)

## Computation of ICER for each random draw

IncrementalC <- ce_sim_part1[, idcosts := dcosts.y - dcosts.x]
IncrementalQ <- ce_sim_part1[, idqalys := dqalys.y - dqalys.x]
meanICER <- ce_sim_part1[, .(icer = mean(idcosts)/mean(idqalys))]

**Reply by n8thangreen** [reply]
agree. I think removing the two-step-ness would be great

**Comment by CKENNEDY-HE** [comment]
I think this is definitely an interesting idea, I just wonder how you would control for lead time bias in this type of two-stage model? Typically cancer screening models, where the benefit of the intervention is earlier diagnosis, would simulate a "tumour genesis time" rather than diagnosis time to control for this. Therefore utilising a DES model approach.

This paper has a great discussion about lead time bias: https://www.nature.com/articles/s41598-021-91852-6

**Reply by yishu-lin** [reply]
I'm interested in handling the lead time bias in the cancer screening model. I think we probably need a more detailed model (with more health states) and do a model calibration.
